CMakeLists.txt

```diff
@@ -1,16 +1,27 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
```

Comment:
minimum version must be `3.19` for this CMakeLists.txt

Reply:
Are you sure?  I didn't think anything here was particularly new.  I certainly don't want to force people to update there system cmake just to build azmq, unless there's a compelling reason.

Reply:
I see your reference
https://cmake.org/cmake/help/v3.21/command/add_library.html?highlight=interface#interface-libraries